velox/vector/arrow/Bridge.cpp

```diff
@@ -327,6 +327,12 @@ void gatherFromBuffer(
       int128_t value = decimalSrc[i].unscaledValue();
       memcpy(dst + (j++) * sizeof(int128_t), &value, sizeof(int128_t));
     });
+  } else if (type.kind() == TypeKind::TIMESTAMP) {
+    rows.apply([&](vector_size_t i) {
+      auto tsSrc = buf.as<Timestamp>();
+      int64_t value = tsSrc[i].toMicros();
```

Comment:
Our use case is microseconds too. It would be nice if the time unit information from the parquet file or the other Arrow data is retained, but maybe it would require too many changes to the velox code base. Maybe an optional parameter could be added to the bridge functions.

Reply:
For parquet file, It can be int96 type or int64 type for micro and milli unit. For Arrow data, there are several different units, including second, milli, micro and nano. But in Velox, there is only one timestamp type composed of seconds and nanos. In this PR, I only support the conversion between Arrow and Velox in micro unit.

Reply:
Hi @rui-mo @zeodtr, I also encountered the problem of the transformation of timestamp type between Velox and Arrow. I propose to add some precision info to timestamp type for data importing and exporting(For internal calculations, there is no need to change the existing data structure). Eg, for a millisecond arrow vector, velox can reads it by fromMillis() and marks it as millis, so we can convert the data to millis precision while exporting, and the same to microsecond and nanosecond.

Reply:
Maybe a config can be added to decide the unit of timestamp conversion between Velox and Arrow. I can extract this part to a separate PR if needed. @Yuhta How do you think?

Reply:
Do you mean a config at the task level or at the vector level?

Reply:
A config kept with vector would only work if you convert the vector from arrow, then without any computation, convert it back to arrow, which is not very useful IMO.  We can add a `Options` object to the bridge interface, and we can get this populate either from query config or connector config.

Reply:
@Yuhta Thank you. Makes sense to me. I can make a prototype for this.

Reply:
@rui-mo Could you notice me if you create a separate PR for this issue, thx.

Reply:
@ViggoC I expect to have a draft PR next week. Will ping you then.

Reply:
@Yuhta @ViggoC Opened https://github.com/facebookincubator/velox/pull/7625 to address the timestamp unit issue. Would you like to review? Thanks.

Reply:
my pleasure